packages/workers-shared/tsconfig.json

```diff
@@ -5,8 +5,8 @@
 		"sourceMap": true,
 		"forceConsistentCasingInFileNames": true,
 		"useUnknownInCatchVariables": false,
-		"types": ["@cloudflare/workers-types"]
+		"types": ["@cloudflare/workers-types/experimental", "vitest/globals"]
```

Comment:
does this look reasonable? alternatively I could have a separate `tsconfig.json` for just the test files, but that felt unnecessary?

Reply:
It's usually good practice to have different tsconfigs for directories which use different types - you'll see it in the fixtures directory a lot, for example

Reply:
kk, thank you for that input <3! I'll adjust

Reply:
I think the way this is donw in `wrangler` is to have both the Vitest config and a separate `tsconfig.json` inside the tests folder. I don't much like having a separate tests folder, but one major benefit is that you can isolate all the test-related config.

Reply:
wrangler has a separate `tsconfig.json` file inside the `__tests__` folder, but `vitest.config.mts` lives in the root. `workers-shared` will most likely share the vitest config between the AssetWorker and RouterWorker, in which case I guess it makes sense for it to live in the root?

Reply:
I went ahead and created a separate `tsconfig.json` file inside the tests folder, which includes the types relevant for tests, and removed `vitest/globals` from the root `tsconfig.json`. I did keep vitest.config in root though for the reasons I explained above. Are we 👌  with that?

Reply:
here we go: https://github.com/cloudflare/workers-sdk/pull/6539/commits/1154d6399a0a989c010fd4df1402ea2b1244e1ce